runtime-modules/constitution/src/lib.rs

```diff
@@ -0,0 +1,83 @@
+//! # Constitution pallet.
+//! `Constitution` pallet for the Joystream platform. Version 1.
+//! It contains current constitution text hash and amendment number in the storage and extrinsic for
+//! setting the new constitution.
+//!
+
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod tests;
+
+use codec::{Decode, Encode};
+use frame_support::{decl_event, decl_module, decl_storage};
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+use sp_runtime::traits::Hash;
+use sp_std::vec::Vec;
+use system::ensure_root;
+
+pub trait Trait: system::Trait {
+    type Event: From<Event> + Into<<Self as system::Trait>::Event>;
+}
+
+/// Contains constitution text hash and its amendment number.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, Default)]
+pub struct ConstitutionInfo {
+    /// Constitution text hash.
+    pub text_hash: Vec<u8>,
+
+    /// Constitution amendment number.
+    pub amendment_number: u32,
```

Comment:
We don't store the history of the amendment numbers. It is always the last one.  This is actual data rather than archival data. Without the 'query node' has more information than the runtime, which is incorrect.

Reply:
I removed the field after our discussion.